cfme/physical/physical_switch.py

```diff
@@ -0,0 +1,231 @@
+            num_sec=1000,
+            fail_func=view.browser.refresh
+        )
+
+    def wait_for_delete(self):
+        """Waits for the Switch to remove from the UI."""
+        view = navigate_to(self.parent, "All")
+        logger.info("Waiting for the Switch to delete...")
+        wait_for(
+            lambda: not self.exists,
+            message="Wait for the Switch to disappear",
+            num_sec=500,
+            fail_func=view.browser.refresh
+        )
+
+
+@attr.s
+class PhysicalSwitchCollection(BaseCollection):
+    """Collection object for the :py:class:`cfme.infrastructure.host.PhysicalSwitch`."""
+
+    ENTITY = PhysicalSwitch
+
+    def select_entity_rows(self, *physical_switches):
+        """ Select all physical Switch objects """
+        physical_switches = list(physical_switches)
+        checked_physical_switches = list()
+        view = navigate_to(self, 'All')
+
+        for physical_switch in physical_switches:
+            view.entities.get_entity(name=physical_switch.name, surf_pages=True).check()
+            checked_physical_switches.append(physical_switch)
+        return view
+
+    def all(self):
+        """returning all physical_switches objects"""
+        physical_switch_table = self.appliance.db.client['switches']
+        ems_table = self.appliance.db.client['ext_management_systems']
+        physical_switch_query = (
+            self.appliance.db.client.session
+                .query(physical_switch_table.name, ems_table.name)
+                .join(ems_table, physical_switch_table.ems_id == ems_table.id))
+
+        provider = self.filters.get('provider')
+        if provider:
+            physical_switch_query = physical_switch_query.filter(ems_table.name == provider.name)
+        physical_switches = []
+        for name, ems_name in physical_switch_query.all():
+            physical_switches.append(self.instantiate(name=name,
+                                    provider=provider or get_crud_by_name(ems_name)))
+        return physical_switches
+
+    def find_by(self, ph_name, provider=None):
+        """returning all physical_switches objects"""
+        physical_switch_table = self.appliance.db.client['switches']
+        ems_table = self.appliance.db.client['ext_management_systems']
+        physical_switch_query = (
+            self.appliance.db.client.session
+                .query(physical_switch_table.name, ems_table.name)
+                .join(ems_table, physical_switch_table.ems_id == ems_table.id))
+
+        if self.filters.get('provider'):
+            provider = self.filters.get('provider')
+            physical_switch_query = physical_switch_query.filter(ems_table.name == provider.name)
+
+        for name, ems_name in physical_switch_query.all():
+            if ph_name == name:
+                return self.instantiate(name=name, provider=provider or get_crud_by_name(ems_name))
+
+
+@navigator.register(PhysicalSwitchCollection)
```

Comment:
**OPTIONAL:** Consider at some point putting in ```am_i_here``` methods to help shortcut navigation steps in the future.

Reply:
@matheuscmelo Could you do this change in another Pull Request?

Reply:
yep, I will

Reply:
Thanks! :D

Reply:
@psav apparently, override the ``am_i_here`` method is unnecessary, since it's already inherited from ``CFMENavigateStep``
https://github.com/ManageIQ/integration_tests/blob/6b09a70d1463ad410c09e46fc27bd0eedb21653f/cfme/utils/appliance/implementations/ui.py#L275-L279

Reply:
Fine :)